src/path_converters.h

```diff
@@ -205,11 +205,9 @@ class PathNanRemover : protected EmbeddedQueue<4>
                    are found along the way, the queue is emptied, and
                    the next curve segment is handled. */
                 code = m_source->vertex(x, y);
-                if (code == agg::path_cmd_stop) {
-                    return code;
-                }
-                if (code == (agg::path_cmd_end_poly | agg::path_flags_close)
-                        && valid_segment_exists) {
+                if (code == agg::path_cmd_stop ||
+                    code == (agg::path_cmd_end_poly | agg::path_flags_close) &&
+                    valid_segment_exists) {
```

Comment:
Because this is multi-line, I would add parentheses even though the precedence is fine.
```suggestion
                    (code == (agg::path_cmd_end_poly | agg::path_flags_close) &&
                     valid_segment_exists)) {
```

Reply:
Repeat below.